packages/react/src/__tests__/ReactChildren-test.js

```diff
@@ -1039,6 +1039,32 @@ describe('ReactChildren', () => {
     });
   });
 
+  // @gate __DEV__
```

Comment:
I removed this line in my latest commit https://github.com/facebook/react/pull/32061/commits/a7b019279a0f66c920c3825706e708132e959e76 as it should work for both dev and prod